tests/records/test_relations_resource_types.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN
+# Copyright (C) 2021 Northwestern University
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Test record relationships."""
+
+import pytest
+from invenio_db import db
+from invenio_records.systemfields.relations import InvalidRelationValue
+
+from invenio_rdm_records.records.api import RDMDraft, RDMRecord
+
+
+#
+# Tests
+#
+def test_resource_types_field(running_app):
+    assert 'resource_type' in RDMDraft.relations
+    assert 'resource_type' in RDMRecord.relations
+    assert RDMDraft.relations.resource_type
+    assert RDMRecord.relations.resource_type
+
+
+def test_resource_type_validation(running_app, minimal_record):
+    """Tests data content validation."""
+    # Valid id
+    minimal_record["metadata"]["resource_type"] = {"id": "image-photo"}
+
+    draft = RDMDraft.create(minimal_record)
+    draft.commit()
+    db.session.commit()
+
+    assert draft["metadata"]["resource_type"] == {"id": "image-photo"}
+
+    # Invalid id
+    minimal_record["metadata"]["resource_type"] = {"id": "invalid"}
+
+    pytest.raises(InvalidRelationValue, RDMDraft.create(minimal_record).commit)
+
+
+def test_resource_types_indexing(running_app, minimal_record):
+    """Test dereferencing characteristics/features really."""
+    minimal_record["metadata"]["resource_type"] = {"id": "image-photo"}
+    draft = RDMDraft.create(minimal_record).commit()
+
+    # TODO/WARNING: draft.dumps() modifies draft
```

Comment:
Is this because of the dumpers? I had the same issue when doing the DataCite serialization... We fixed by making a deep cope [here](https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/dumpers/base.py#L30)... But I see the dumpers in `rdm-records` still modify it... shall we open an issue?